plat/kvm/x86/multiboot2.c

```diff
@@ -0,0 +1,208 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2024, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#include <uk/essentials.h>
+#include <uk/arch/limits.h>
+#include <uk/arch/types.h>
+#include <uk/arch/paging.h>
+#include <uk/plat/bootstrap.h>
+#include <uk/plat/common/bootinfo.h>
+#include <uk/plat/common/lcpu.h>
+#include <uk/plat/common/memory.h>
+#include <uk/plat/common/sections.h>
+#include <uk/reloc.h>
+#include <kvm-x86/multiboot2.h>
+
+#include <errno.h>
+#include <string.h>
+
+#define multiboot2_crash(msg, rc)	ukplat_crash()
+
+void _ukplat_entry(struct lcpu *lcpu, struct ukplat_bootinfo *bi);
+
+static inline void mrd_insert(struct ukplat_bootinfo *bi,
+			      const struct ukplat_memregion_desc *mrd)
+{
+	int rc;
+
+	if (unlikely(mrd->len == 0))
+		return;
+
+	rc = ukplat_memregion_list_insert(&bi->mrds, mrd);
+	if (unlikely(rc < 0))
+		multiboot2_crash("Cannot insert bootinfo memory region", rc);
+}
+
+static void multiboot2_init_mem(struct ukplat_bootinfo *bi,
+				struct multiboot_tag *tag)
+{
+	struct multiboot_tag_mmap *mmap_tag = (struct multiboot_tag_mmap *)tag;
+
+	struct multiboot_mmap_entry *mmap = mmap_tag->entries;
+	struct ukplat_memregion_desc mrd = {0};
+	__paddr_t start, end;
+	__sz offset;
+
+	__sz mmap_entry_cnt = (mmap_tag->size - sizeof(struct multiboot_tag_mmap)) /
+			    		 mmap_tag->entry_size;
+
+	for (offset = 0; offset < mmap_entry_cnt;
+		offset++) {
```

Comment:
This is also misaligned but it will be unproductive if I keep spamming for each single coding style issue. I would advise using `support/scripts/checkpatch.pl --strict` on your patches and solve the warnings/errors.